torch/_inductor/codegen/triton.py

```diff
@@ -1161,11 +1162,27 @@ def __init__(self, line, var, mask, size_map):
                 self.size_map = size_map
 
             def __call__(self):
-                # The conditions need to be in parens because of Python's operator precedence.
-                # It'd be less # error-prone to use and/or/not, which is suported by triton
-                size = self.size_map[(self.var, self.mask)]
-                cond = f"(0 <= {self.var}) & ({self.var} < {size})"
-                cond_print = f"0 <= {self.var} < {size}"
+                size, size_str = self.size_map[(self.var, self.mask)]
+
+                assert_min = (var.bounds.lower < 0) == sympy.true
+                assert_max = (var.bounds.upper > size) == sympy.true
```

Comment:
I take it this is `sympy.Lt` for dynamic shapes, so `sympy.Lt != sympy.true` and we generate an assert?

```suggestion
                assert_min = (self.var.bounds.lower < 0) == sympy.true
                assert_max = (self.var.bounds.upper >= size) == sympy.true
```

Reply:
I'm assuming `bound.upper` is a closed range `[lower, upper]` but if it's half open then `> size` is correct.

Reply:
https://github.com/pytorch/pytorch/pull/104987. Funnily enough, this issue has been there since the inception of these bounds. How about that.